service.go

```diff
@@ -285,7 +294,44 @@ func (s *service) push(
 				return
 			}
 
-			log.Debugf("reply from %s: %t", p, reply.Ok)
+			log.Debugf("received reply from %s", p)
+
+			// Handle new log addresses
+			if reply.NewAddr != nil {
+				err = s.threads.AddAddr(id, lid, reply.NewAddr.Multiaddr, pstore.PermanentAddrTTL)
```

Comment:
Replaying what happened up to this point to validate:
- We want to push a record to a log
- We send it to all known log addresses
- Every `PushReply` now has a `newAddr` 
- If `newAddr` has value, at this quoted line we add it as a new address to `id, lid` which is basically is adding (or refreshing) the _confirmed_ multiaddr for future pulls from the log for other peers (or whatever addrs are needed for this log). Addresses from owners and followers are undistinguishable for the books, right?

The `newAddr` field being not nil is valid for every possible type of record?

Reply:
That sounds correct to me! I'm not in love with the `newAddr` response, perhaps we could specialize messages further with a `Push` and `PushToFollower` message type / API at some point.